model/task/task.go

```diff
@@ -1632,3 +1635,62 @@ func (t *Task) GetJQL(searchProjects []string) string {
 
 	return fmt.Sprintf(jqlBFQuery, strings.Join(searchProjects, ", "), jqlClause)
 }
+
+// BlockedState returns "blocked," "pending," or "" to represent the state of the task
+// with respect to its dependencies
+func (t *Task) BlockedState() (string, error) {
+	if t.DisplayOnly {
+		return t.blockedStateForDisplayTask()
+	}
+
+	dependencyIDs := []string{}
+	for _, d := range t.DependsOn {
+		dependencyIDs = append(dependencyIDs, d.TaskId)
+	}
+	dependentTasks, err := Find(ByIds(dependencyIDs).WithFields(DisplayNameKey, StatusKey,
+		ActivatedKey, BuildVariantKey, DetailsKey, DependsOnKey))
+	if err != nil {
+		return "", errors.Wrap(err, "error finding dependencies")
+	}
+	taskMap := map[string]*Task{}
+	for _, t := range dependentTasks {
+		taskMap[t.Id] = &t
```

Comment:
You are taking the address of the task that was copied by the range operator here. This won't work as you expect.

Reply:
fixed